edit/edit.go

```diff
@@ -412,14 +416,15 @@ func sub(ed *Editor, a Address, re *re1.Regexp, repl []rune, g bool) (addr, erro
 			break
 		}
 		from = m[0][1]
+		n = 1 // reset n to 1, so that on future iterations of this loop we get the next instance.
 	}
 	return at, nil
 }
 
-// Sub1 substitutes the first match of the regular expression
+// SubSingle substitutes the Nth match of the regular expression
 // with the replacement specifier.
-func sub1(ed *Editor, at addr, re *re1.Regexp, repl []rune) ([][2]int64, error) {
-	m, err := match(ed, at, re)
+func subSingle(ed *Editor, at addr, re *re1.Regexp, repl []rune, n int) ([][2]int64, error) {
```

Comment:
Nevermind. It's just because github has a terrible code review UI. It collapsed 5 lines of this function, so I thought it was much shorter.